### ISSUE TITLE
Enable two-level initialization by default

### ISSUE THREAD

**Issue by uekerman** [precice/precice#633]
There are still a few details that need to be sorted out before we can enable two-level intitialization by default.

* [ ] `ReceivedPartition::createOwnerInformation` still applies a gather-scatter approach and is therefore a potential bottleneck
* [ ] Move exchange of `communicationMap` out of `Partition::compute()`. Probably to `m2n::completeSlavesConnection()`. Otherwise, we might get blocking sends as `MeshContexts` are in wrong order when calling `partition::compute`.
* [ ] Merge `accept/requestSlaves` and `completeSlavesConnection` to a single entry point
* [ ] Extend two-level initialization to multiple m2ns in `ProvidedPartition`
* [ ] Support coupling mode (i.e. a serial participant) with two-level initialization or automatically switch
* [ ] Support gather-scatter m2n with two-level initialization or automatically switch
* [ ] Get rid of vertex distribution and use only local information (`connectedRanks` or `communicationMaping`) in `m2n`. Reduce code duplication in `PointToPointCommunication`. 

Additionally, we need a better test coverage.

**Comment by BenjaminRodenberg** [comment]
What's the status here? I guess it would be nice to use two-level initialization as the default option and delete the alternative implementation with the breaking release, because it should simplify the code. But I have no idea whether this is actually realistic. The todo list above looks like it is not.

**Comment by uekerman** [comment]
No, that's unfortunately unrealistic. I guess that most of the points above are still open.
Independent of that, I would not delete the old gather-scatter mode as it is helpful for debugging.
I see this issue also not as breaking. It will only be faster :)
